### ISSUE TITLE
Trees/Archives with missing change log notes for the current tag.

### ISSUE THREAD

**Issue by ctubio** [skywinder/github-changelog-generator#230]
Hello @skywinder !
very nice changelog generator :heart_eyes:
Maybe im not understanding something, but to me looks like something is *wrong*.

Using your lovely project as example, if you go to tree 1.3.9, in its changelog, it only appears notes until 1.3.8, and you need to go to 1.3.10 to be able to read the notes of 1.3.9.

See change log file of tree 1.3.9 with missing change log notes for 1.3.9 at https://github.com/skywinder/github-changelog-generator/blob/1.3.9/CHANGELOG.md

The real *wrong* thing here for my eyes, is that the archives for 1.3.9 misses the change log notes for 1.3.9 as seen at https://github.com/skywinder/github-changelog-generator/archive/1.3.9.tar.gz

Obviously, this occurs because the generation of the change log file happens to be after the creation of the tag, and therefore, the commit of the change log notes happens to fall into the next tag.

But sadly, it stills looks to me that each tree/archive should contain its own change log notes.

Did you thought about it?

TL:DR: aww very niiicee change log generator! can we include change log notes of the current tag in their respective archives/trees *somehow*?

**Comment by skywinder** [comment]
Hi! Thanks for mention it. Yes, You're right, now there is no way to build proper change log before setting a tag.
If you look at the git log of this project - you will find, right after setting the new tag the next commit is updates change log file.
It's a bit confusing, but I have no idea, which tag will be next.

What do you think, how better to solve this issue?

**Comment by skywinder** [comment]
The only workaround, that I see - is add ability to manually set label for `Unreleased` section. 
Then, you can generate correct change log before tagging.

For example implement such options:

      github_changelog_generator --unreleased-label 1.4.0

That will change unreleased section name to 1.4.0. 
Because `Unrelesed` link always look for the `HEAD`, (https://github.com/skywinder/github-changelog-generator/compare/1.3.11...HEAD) there is another option:

      github_changelog_generator --next-tag 1.4.0


To change not only label, but also link for section (that not exists yet). 
Like:
(https://github.com/skywinder/github-changelog-generator/compare/1.3.11...1.4.0)

**Comment by ctubio** [comment]
ouYea, that is more than a workaround i may i say, its a lovely feature to avoid the github's workflow intrinsics xD thank you for sharing your brain! (i was stucked there xD)

(i will try to make a pull request but im not a ruby expert, so.. if you wanna go for it, i will not feel bad to see that you were faster xD at least i learned some lovely ruby meanwhile)

thanks again for your time!

**Comment by skywinder** [comment]
@ctubio Yep, ruby is awesome.
bdw, already implemented in @231. :wink:

**Comment by ctubio** [comment]
lets close this issue in favor of #231

thanks alot guys (or penguin)
byee''
